web/src/engine/common/web-utils/src/kmwstring.ts

```diff
@@ -1,453 +1,412 @@
+  keyboards exist that utilize characters within such ranges.
 ***/
 
+/**
+ * Indicates if functionality supporting Supplementary Multilingual Plane (SMP)
+ * characters should be enabled.
+ */
+let EXTENSION_ENABLED = false;
 
-/*
- * TODO:  Remove this file as part of addressing https://github.com/keymanapp/keyman/issues/2492.
+/**
+ * Enables or disables support for SMP-aware string handling.
+ * @param bEnable
  */
+export function enableSupplementaryPlane(bEnable: boolean) {
+  EXTENSION_ENABLED = bEnable;
+}
 
-declare global {
-  interface StringConstructor {
-    kmwEnableSupplementaryPlane(bEnable: boolean): void
-  }
-
-  interface String {
-    kmwCharCodeAt(codePointIndex: number): number,
-    kmwCharAt(codePointIndex: number) : string,
-    kmwIndexOf(searchValue: string, fromIndex?: number) : number,
-    kmwLastIndexOf(searchValue: string, fromIndex?: number) : number,
-    kmwSlice(beginSlice: number, endSlice: number) : string,
-    kmwSubstring(start: number, length: number) : string,
-    kmwSubstr(start: number, length?: number) : string,
-    kmwBMPSubstr(start: number, length?: number) : string,
-    kmwLength(): number,
-    kmwBMPLength(): number,
-    kmwNextChar(codeUnitIndex: number): number,
-    kmwBMPNextChar(codeUnitIndex: number): number,
-    kmwPrevChar(codeUnitIndex: number): number,
-    kmwBMPPrevChar(codeUnitIndex: number): number,
-    kmwCodePointToCodeUnit(codePointIndex: number) : number,
-    kmwBMPCodePointToCodeUnit(codePointIndex: number) : number,
-    kmwCodeUnitToCodePoint(codeUnitIndex: number) : number,
-    kmwBMPCodeUnitToCodePoint(codeUnitIndex: number) : number,
-    _kmwCharCodeAt(codePointIndex: number): number,
-    _kmwCharAt(codePointIndex: number) : string,
-    _kmwIndexOf(searchValue: string, fromIndex?: number) : number,
-    _kmwLastIndexOf(searchValue: string, fromIndex?: number) : number,
-    _kmwSlice(beginSlice: number, endSlice: number) : string,
-    _kmwSubstring(start: number, length?: number) : string,
-    _kmwSubstr(start: number, length?: number) : string,
-    _kmwLength(): number,
-    _kmwNextChar(codeUnitIndex: number): number,
-    _kmwPrevChar(codeUnitIndex: number): number,
-    _kmwCodePointToCodeUnit(codePointIndex: number) : number,
-    _kmwCodeUnitToCodePoint(codeUnitIndex: number) : number
+/**
+ * Returns a number indicating the Unicode value of the character at the given
+ * code point index, with support for supplementary plane characters.
+ *
+ * @param  {string}  s      The string
+ * @param  {number}  index  The code point index into the string (not
+                            the code unit index) to return
+ * @return {number}         The Unicode character value
+ */
+export function charCodeAt(s: string, index: number) {
+  // While String.codepointAt exists, it does not use code-point indexing and so is
+  // unsuited to our needs here.
+  const str = s;
+
+  if(!EXTENSION_ENABLED) {
+    return s.charCodeAt(index);
   }
```

Comment:
No more twin methods / method swapping.  Instead, we just guard early within each method's implementation, using the simple approach when SMP handling is disabled and passing through to the SMP stuff when it isn't disabled.